examples/python/mp_face/main.py

```diff
@@ -0,0 +1,419 @@
+    """
+    Logger for the MediaPipe Face Landmark Detection solution.
+
+    https://developers.google.com/mediapipe/solutions/vision/face_landmarker
+    """
+
+    MODEL_PATH: Final = (MODEL_DIR / "face_landmarker.task").resolve()
+    MODEL_URL: Final = (
+        "https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/latest/"
+        "face_landmarker.task"
+    )
+
+    def __init__(self, video_mode: bool = False, num_faces: int = 1):
+        self._video_mode = video_mode
+
+        # download model if necessary
+        if not self.MODEL_PATH.exists():
+            download_file(self.MODEL_URL, self.MODEL_PATH)
+
+        self._base_options = mp.tasks.BaseOptions(
+            model_asset_path=str(self.MODEL_PATH),
+        )
+        self._options = vision.FaceLandmarkerOptions(
+            base_options=self._base_options,
+            output_face_blendshapes=True,
+            num_faces=num_faces,
+            running_mode=mp.tasks.vision.RunningMode.VIDEO if self._video_mode else mp.tasks.vision.RunningMode.IMAGE,
+        )
+        self._detector = vision.FaceLandmarker.create_from_options(self._options)
+
+        # Extract classes from MediaPipe face mesh solution. The goal of this code is:
+        # 1) Log an annotation context with one class ID per facial feature. For each class ID, the class description
+        #    contains the connections between corresponding keypoints (taken from the MediaPipe face mesh solution)
+        # 2) A class ID array matching the class IDs in the annotation context to keypoint indices (to be passed as
+        #    the `class_ids` argument to `rr.log_points`).
+
+        classes = [
+            mp.solutions.face_mesh.FACEMESH_LIPS,
+            mp.solutions.face_mesh.FACEMESH_LEFT_EYE,
+            mp.solutions.face_mesh.FACEMESH_LEFT_IRIS,
+            mp.solutions.face_mesh.FACEMESH_LEFT_EYEBROW,
+            mp.solutions.face_mesh.FACEMESH_RIGHT_EYE,
+            mp.solutions.face_mesh.FACEMESH_RIGHT_EYEBROW,
+            mp.solutions.face_mesh.FACEMESH_RIGHT_IRIS,
+            mp.solutions.face_mesh.FACEMESH_FACE_OVAL,
+            mp.solutions.face_mesh.FACEMESH_NOSE,
+        ]
+
+        self._class_ids = [0] * mp.solutions.face_mesh.FACEMESH_NUM_LANDMARKS_WITH_IRISES
+        class_descriptions = []
+        for i, klass in enumerate(classes):
+            # MediaPipe only provides connections for class, not actual class per keypoint. So we have to extract the
+            # classes from the connections.
+            ids = set()
+            for connection in klass:
+                ids.add(connection[0])
+                ids.add(connection[1])
+
+            for id_ in ids:
+                self._class_ids[id_] = i
+
+            class_descriptions.append(
+                rr.ClassDescription(
+                    info=rr.AnnotationInfo(id=i),
+                    keypoint_connections=klass,
+                )
+            )
```

Comment:
Do you have any thoughts on how to improve "AnnotationContext" apis to make this easier to implement?

Reply:
My "gut feeling"-level comprehension is currently as follows:
- Although its scope is currently narrow, annotation contexts are meant to be a somewhat generic feature to attach semantics (that the viewer understands) to specific entities.
- ClassDescription is _a_ specific annotation: that attaches connection/colour semantics to the "keypoint" component of points2d/3d
- AnnotationInfo is a (potentially generic?) low-level data structure used for storage of annotations such as ClassDescription.

If that's a more or less correct interpretation (that's a big "if"!), then:
- "Context" is both loaded and vague. We should try to find a better naming. "Annotation" (without context) might just be good, provided that the notion of "Annotation" is properly documented in "Concepts".
- "ClassDescription" could have a better name as well. It pertains to keypoint ids, so its name should probably refer to that? TBD.
- "AnnotationInfo" is a leaking implementation detail, and should disappear altogether from the API. "ClassDescription" should have a better, more explicit API instead. I guess this could be addressed by HOPE.

I'm not sure about how to best address the specific situation here, but I probably don't want to think too much about it until what precedes is sorted out in my mind :)

Reply:
Makes sense, could you open an issue on this? I think this is a powerful feature that is being underused at the moment